rkt/stage1hash.go

```diff
@@ -287,9 +287,17 @@ func getConfiguredStage1Hash(s *store.Store, imgRef, imgLoc, imgFileName string)
 	return getStage1HashFromPath(fn, imgLoc, imgFileName)
 }
 
-func getStage1Finder(s *store.Store) *image.Finder {
+func getStage1Finder(s *store.Store, location string) *image.Finder {
+	ks := getKeystore()
+	// if the location of stage1 is in the default directory, we don't check
+	// the signature.
+	absLocation := filepath.Clean(location)
+	if strings.HasPrefix(absLocation, buildDefaultStage1ImagesDir) {
```

Comment:
Replace `buildDefaultStage1ImagesDir` with `fmt.Sprintf("%s%c", filepath.Clean(buildDefaultStage1ImagesDir), filepath.Separator)`. This is to avoid skipping signature validation when `buildDefaultStage1ImagesDir` is `/usr/lib/rkt/stage1-images` and `absLocation` is /usr/lib/rkt/stage1-images-haha-in-your-face/gimme-your-login.aci`.